### PR TITLE
Implementation of coupledVectorDotDu and coupledVectorDotDotDu

### LINKED ISSUE #13913: coupledVectorDotDu missing for Vector Variables
## Reason
When hand-coding Jacobians using vector variables, `coupledVectorDotDu` is needed. Currently, only `coupledVectorDot` exists. Due to ELK performing std::complex operations in several BCs, AD is not currently an option for those Jacobians (since taking the finite difference derivative w.r.t a complex variable can lead to multiple outcomes, depending on what direction in the complex plane is chosen for the derivative - thus, the AD system currently sets these derivatives equal to `NaN` and terminates).

## Design
The design of this function should mirror that of `dotDu` but with vector types.

## Impact
Adds new capability for cases when hand-coding a Jacobian for a vector object is still needed.

### PR DESCRIPTION
**Pull request by cticenhour**
Mirrors that of the scalar classes, but with vector variables. Adds:

- `coupledVectorDotDu` method in `Coupleable`
- `coupledVectorDotDotDu` method in `Coupleable`
- `VectorTimeKernel` class
- `VectorTimeDerivative` class
- `CoupledVectorTimeDerivative` class
- Tests for new derivative classes, and documentation for both.

Closes #13913 

Requesting a review from @lindsayad, since we've previously discussed this PR offline. Alex, this ended up being a little simpler than I originally implemented it as, since the `du_dot_du` applied in the Jacobian statement is still a scalar value, regardless of whether the variable is a vector or not. So this implementation is able to avoid all of that type templating that I originally thought I needed to do. One more thing I need to add is a test that touches `coupledVectorDotDotDu`, so I guess I should pull inspiration from one of the modules - a code search told me that its only being used in the framework for `CoupledValuesMaterial`. I am open to suggestions here, but I thought this PR was ready for an initial review.

### CODE REVIEW

**Comment by cticenhour** [comment]
@lindsayad I believe this is ready for a formal review.

File: test/tests/kernels/transient_vector_diffusion/tests [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[Tests]
+  [./transient_vector_diffusion]
+    type = 'Exodiff'
+    input = 'transient_vector_diffusion.i'
+    exodiff = 'transient_vector_diffusion_out.e'
+    requirement = 'We shall be able to reproduce the simple_transient_diffusion results with a vector compatible TimeDerivative (VectorTimeDerivative) containing a hand-coded Jacobian.'
```
Comment (lindsayad) [inline test/tests/kernels/transient_vector_diffusion/tests]: @permcody has been trying to get us to be better at how we do requirements. See his comments [here](https://github.com/idaholab/moose/pull/13944#discussion_r316912819) and [here](https://github.com/idaholab/moose/pull/13944#discussion_r316913366)

File: test/tests/kernels/vector_dot_dot/vector_test.i [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# Tests calculation of first and second time derivative
+# of a coupled vector variable in a material
+# a_vec(x,y,z,t) = [t*(t*x + y), t*y, 0]
+# a_vec_dot(x,y,z,t) = [2*t*x + y, y, 0]
+# a_vec_dot_dot(x,y,z,t) = [2*x, 0, 0]
+#
+# IMPORTANT NOTE:
+# Currently, this test produces a_vec_dot and a_vec_dot_dot that contains oscillations over time.
+# This is a known by-product of Newmark Beta time integration (see the Newmark Beta documentation),
+# but as of Summer 2019, there is no alternative time integrator in MOOSE that can contain these
```
Comment (lindsayad) [inline test/tests/kernels/vector_dot_dot/vector_test.i]: Did you mean dampen instead of contain at the end of this line?

**Review by lindsayad — CHANGES_REQUESTED** [review]
Looks excellent to me. Just the couple small comments

File: test/tests/kernels/transient_vector_diffusion/tests (hunk 1 above)
Comment (cticenhour) [reply test/tests/kernels/transient_vector_diffusion/tests]: Shortened and modified the requirement to (hopefully) better fit the guidelines.

File: test/tests/kernels/vector_dot_dot/vector_test.i (hunk 2 above)
Comment (cticenhour) [reply test/tests/kernels/vector_dot_dot/vector_test.i]: Yes! Thanks for the catch.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**